extended-tracer/src/main/java/io/opentelemetry/contrib/tracer/Tracing.java

```diff
@@ -0,0 +1,269 @@
+        @Nullable
+        public String get(@Nullable Map<String, String> carrier, String key) {
+          //noinspection ConstantConditions
+          return carrier == null ? null : carrier.get(key);
+        }
+      };
+
+  private Tracing() {}
+
+  public static void run(String spanName, Runnable runnable) {
+    serviceTracer().run(spanName, runnable);
+  }
+
+  public static void run(Span span, Runnable runnable) {
+    call(
+        span,
+        () -> {
+          runnable.run();
+          return null;
+        });
+  }
+
+  /**
+   * Runs a block of code with a new span - ending the span at the end and recording any exception.
+   *
+   * @param spanName name of the new span
+   */
+  public static <T> T call(String spanName, Callable<T> callable) {
+    return serviceTracer().call(spanName, callable);
+  }
+
+  public static <T> T call(Span span, Callable<T> callable) {
+    return call(span, callable, Tracing::setSpanError);
+  }
+
+  @SuppressWarnings("NullAway")
+  public static <T> T call(
+      Span span, Callable<T> callable, BiConsumer<Span, Exception> handleException) {
+    //noinspection unused
+    try (Scope scope = span.makeCurrent()) {
+      return callable.call();
+    } catch (Exception e) {
+      handleException.accept(span, e);
+      sneakyThrow(e);
+      return null;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Return the tracer to be used in a service
+   *
+   * @return the tracer to be used in a service
+   */
+  public static ExtendedTracer serviceTracer() {
+    return ExtendedTracer.create(GlobalOpenTelemetry.getTracer("service"));
+  }
+
+  /**
+   * Creates a new span builder with the given span name.
+   *
+   * @param spanName the span name
+   * @return the span builder
+   */
+  public static ExtendedSpanBuilder withSpan(String spanName) {
+    return Tracing.serviceTracer().spanBuilder(spanName);
+  }
+
+  /**
+   * Marks a span as error.
+   *
+   * @param span the span
+   * @param description what went wrong
+   */
+  public static void setSpanError(Span span, String description) {
+    span.setStatus(StatusCode.ERROR, description);
+  }
+
+  /**
+   * Marks a span as error.
+   *
+   * @param span the span
+   * @param exception the exception that caused the error
+   */
+  public static void setSpanError(Span span, Throwable exception) {
+    span.setStatus(StatusCode.ERROR);
+    span.recordException(exception);
+  }
+
+  /**
+   * Marks a span as error.
+   *
+   * @param span the span
+   * @param description what went wrong
+   * @param exception the exception that caused the error
+   */
+  public static void setSpanError(Span span, String description, Throwable exception) {
+    span.setStatus(StatusCode.ERROR, description);
+    span.recordException(exception);
+  }
+
+  /**
+   * Injects the current context into a string map, which can then be added to HTTP headers or the
+   * metadata of an event.
+   */
+  public static Map<String, String> injectContext() {
```

Comment:
`injectContext` sounds like it's doing the injection, instead of just returning the headers that should be injected

Reply:
good point - what about `getPropagationHeaders`?